grafana/k6-studio
# feat: enable usage report

## Description

This PR introduces changes to collect [usage report data when running a k6 script](https://grafana.com/docs/k6/latest/set-up/usage-collection/) as well as the following when running k6 Studio:

- os
- arch
- version
- timestamp

The user can opt-out of the usage report by disabling it in the Settings dialog.



## How to Test

- Check that `usageReport.enabled` property is properly saved
- Run a script from k6 Studio and check that the option `--no-usage-report` is passed to the k6 binary (you can do that by logging the `k6Args` array in the `script.ts` file.



## Checklist



## Screenshots (if appropriate):

<img width="822" alt="image" src="https://github.com/user-attachments/assets/1499511d-a9f7-4b6f-9c76-bd457d460efd">


## Related PR(s)/Issue(s)
